File: 0019-remove-nth-node-from-end-of-list/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode curr=head;
        int length=0;
        while(curr!=null)
        {
            length++;
            curr=curr.next;
        }
        int index=(length-n);
            if (index == 0) {
            return head.next; 
        }
        ListNode temp=head;
        ListNode prev=null;
        for(int i=0;i<index;i++)
        {
            prev=temp;
            temp=temp.next;
            
        }
        prev.next=temp.next;
        return head;
    }
}

File: 0019-remove-nth-node-from-end-of-list/solution.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* removeNthFromEnd(struct ListNode* head, int n) {
    int c=0;
    struct ListNode* temp1=head;
    while(temp1!=NULL)
    {
        c++;
        temp1=temp1->next;
    }
    int pos=c-n;
    if(pos==0)
        return head->next;
    struct ListNode*  temp=head;
    struct ListNode* prev=NULL;
    for(int i=0;i<pos;i++)
    {
        prev=temp;
        temp=temp->next;
    }
    prev->next=temp->next;
    free(temp);
    return head;
}
